06week/higherOrder.js

```diff
@@ -2,32 +2,58 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
-}
+const myForEach = (arr, callback) => {
 
-function map(arr, callback) {
-  // Your code here
-}
+  for (let i = 0; i < arr.length; i++) {
+    callback();
+  }
+};
 
-function filter(arr, callback) {
-  // Your code here
-}
 
-function some(arr, callback) {
-  // Your code here
+const map = (arr, callback) => {
+  const mapFormatted = []
+  arr.forEach((item) => {
+    callback(item);
+    mapFormatted.push(callback(item));
+    // console.log(callback(item));
+  });
+  return mapFormatted;
+};
+
+const filter = (arr, callback) => {
+  const filterFormatted = [];
+  for (let j = 0; j < arr.length; j++) {
+    if (arr[j] % 2 === 0) {
```

Comment:
why does it matter if it's even? Your filter should only push the item if the callback returns a truthy value. Filter has nothing to do with even numbers